src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -369,25 +365,36 @@ private fun HearingDto.toNomisUpdateHearing(): UpdateHearingRequest = UpdateHear
   internalLocationId = this.locationId,
 )
 
-private fun HearingOutcomeDto.toNomisCreateHearingResult(
-  adjudicationStatus: String,
-  hearingType: HearingDto.OicHearingType,
-): CreateHearingResultRequest =
-  CreateHearingResultRequest(
-    pleaFindingCode = this.plea!!.name,
-    findingCode = toNomisFindingCode(adjudicationStatus),
-    adjudicatorUsername = getAdjudicatorUsernameForInternalHearingOnly(hearingType),
+// logic for determining the NOMIS findingCode:
+// If separate outcome block exists use code, else use code from hearing.outcome
+private fun OutcomeHistoryDto.toNomisCreateHearingResult(): CreateHearingResultRequest {
+  val separateOutcome: OutcomeDto? = this.outcome?.outcome
+  val hearing = this.hearing!!
+  val findingCode = separateOutcome?.code?.name ?: hearing.outcome!!.code.name
```

Comment:
As the structure of outcomes change considerably depending on the hearing type - we don't support for mandatory fields hence all the !!
I'll look at improving this once the dtos from adjudications have stabilised